retrieval/scrape.go

```diff
@@ -547,13 +568,14 @@ loop:
 		}
 
 		mets := yoloString(met)
-		ref, ok := sl.cache[mets]
+		ref, ok := sl.refCache[mets]
 		if ok {
 			switch err = app.AddFast(ref, t, v); err {
 			case nil:
+				samplesScraped[sl.lsetCache[ref].str] = sl.lsetCache[ref].lset
 			case storage.ErrNotFound:
 				ok = false
-			case errSeriesDropped:
+			case errSeriesDropped, storage.ErrOutOfOrderSample, storage.ErrDuplicateSampleForTimestamp:
```

Comment:
Can we add back logging for the out-of-order and duplicate-sample cases? I think there's not eveng a metric for it anymore at the moment.